### PR TITLE
Refactor AWS ecs_ecr module to rely on AnsibleAWSModule

### PR DESCRIPTION
**Pull request by maastiff**
##### SUMMARY
This PR consists in refactoring AWS ecs_ecr module to rely on AnsibleAWSModule helper, following @tremble ask for improvement in #62892

This PR follow this guide in order to migrate this module accordingly.
https://docs.ansible.com/ansible/devel/dev_guide/platforms/aws_guidelines.html#porting-code-to-ansibleawsmodule

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
ecs_ecr

##### ADDITIONAL INFORMATION
This PR include a BREAKING CHANGE in [https://github.com/ansible/ansible/commit/8207143fc00573f81efa0015d9ee16d834ae12f0](this commit) leveraging camel-dict-to-snake-dict helper which lead to a breaking change in the result format.

see: https://docs.ansible.com/ansible/devel/dev_guide/platforms/aws_guidelines.html#camel-dict-to-snake-dict

### CODE REVIEW

**Comment by Maastiff** [comment]
Hi @tremble 

I just submit this PR as discussed in #62892 to improve ecs_ecr module. Several questions regarding this:
- I had to rely on `module._module._verbosity` but that seems a bit awkward, is it acceptable? (see: [https://github.com/ansible/ansible/pull/62985/files#diff-e324173cfb6811876e47cbf5f82c0dbbL372](l372)
- I rely on `camel_dict_to_snake_dict` to comply with ansible guidelines but this would lead to breaking change for existing users relying to `result.repository`. Do I have to mention this breaking change somewhere?
- Do you have guidelines (or example module) which I could inspire from in order to improve consistency between ecs_ecr and others AWS modules?
- Do you see any other room of improvement?

Best regards,

**Comment by tremble** [comment]
I just removed the ecs_ecr.py lines from test/sanity/ignore.txt which shows up a couple more niggles:

```
ERROR: Found 4 validate-modules issue(s) which need to be resolved:
ERROR: lib/ansible/modules/cloud/amazon/ecs_ecr.py:0:0: doc-missing-type: Argument 'name' in argument_spec uses default type ('str') but documentation doesn't define type
ERROR: lib/ansible/modules/cloud/amazon/ecs_ecr.py:0:0: doc-missing-type: Argument 'registry_id' in argument_spec uses default type ('str') but documentation doesn't define type
ERROR: lib/ansible/modules/cloud/amazon/ecs_ecr.py:0:0: doc-missing-type: Argument 'state' in argument_spec uses default type ('str') but documentation doesn't define type
ERROR: lib/ansible/modules/cloud/amazon/ecs_ecr.py:0:0: parameter-type-not-in-doc: Argument 'policy' in argument_spec defines type as 'json' but documentation doesn't define type
See documentation for help: https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/validate-modules.html
Running sanity test 'yamllint' with Python 3.6
ERROR: The 1 sanity test(s) listed below (out of 53) failed. See error output above for details.
validate-modules
```

File: lib/ansible/modules/cloud/amazon/ecs_ecr.py [hunk 1]
```diff
@@ -350,31 +344,26 @@ def run(ecr, params, verbosity):
...
+        supports_check_mode=True,
+        mutually_exclusive=[['policy', 'delete_policy']])
 
-    ecr = EcsEcr(module)
-    passed, result = run(ecr, module.params, module._verbosity)
+    passed, result = run(EcsEcr(module), module.params, module._module._verbosity)
 
     if passed:
-        module.exit_json(**result)
+        module.exit_json(**camel_dict_to_snake_dict(result))
```
Comment (tremble) [inline lib/ansible/modules/cloud/amazon/ecs_ecr.py:364]: @s-hertel / @jillr : Technically this is a breaking change, but it brings things in line with the Python/Ansible norms and the module is flagged as 'preview' thoughts?

**Comment by tremble** [comment]
You already found the only doc I know of WRT consistency.

As far as verbosity is concerned: the only AWS module using verbosity is ecs_ecr, personally I'd recommend removing it and just consistently returning all of the values, doing otherwise could actually result in playbooks which work only in verbose mode which would be extremely confusing.  (again willing to be overruled by folks like jillr/s-hertel who are far more familiar with the code than I am)

File: lib/ansible/modules/cloud/amazon/ecs_ecr.py (hunk 1 above)
Comment (maastiff) [reply lib/ansible/modules/cloud/amazon/ecs_ecr.py:364]: Indeed, it's flagged preview :)

**Comment by jillr** [comment]
I like the idea of making the return values consistent with Ansible norms and it helps that the `ecs_*` modules are all in preview, however most of the `ecs_*` modules currently return camel case.  If we were to change it we should change all of them in a separate PR where we can discuss possible impacts and get them all covered with a single [porting guide](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.10.rst) entry.

The rest of this change lgtm, thanks @Maastiff.

**Comment by s-hertel** [comment]
Regardless of it being preview, especially since this module has been around since 2.3, we should try to have a deprecation period and a porting guide like @jillr said so people have a chance to update their playbooks before it breaks. There's unfortunately not a great way to deprecate return values (there needs to be a way to opt in to the new behavior to get rid of those warnings). Some modules have simply adding the conventional snake_case options in addition to the legacy CamelCase. In other cases a new module (or a new module name) has been used to return the new data when ported to boto3 and the old module has just been deprecated. Here's one of the most recent examples of the return data changing for a preview module via a rename https://github.com/ansible/ansible/commit/2d98734ad51c3de557c3a7d1f7166bdd114ceb0f#diff-076fbee0e72cc8b32df04fb66e17cf0e.

**Comment by tremble** [comment]
This PR seems to have stalled out because of the changes to the return codes.

The core refactor would be
resolved_by_pr #67191

**Comment by tremble** [comment]
close_me

**Pull request closed without merging**